pkg/mount/mount.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mount
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/errwrap"
+)
+
+// GetMountsForPrefix returns all mountpoints associated with a process mount namespace,
+// filtered by prefix. If no prefixPath is specified, all mounts are returned.
+// The special value 0 as pid argument is used to specify the current process.
+func GetMountsForPrefix(prefixPath string, pid uint) (Mounts, error) {
```

Comment:
We could go super-fancy explicit:

1. Let `mountinfo.Mounts() (Mounts, error)` return all mounts.

2. Introduce:
```
package mountinfo

// FilterFunc is a functional type which returns true if the given Mount should be filtered, else false.
type FilterFunc func(m *Mount) bool

// HasPrefix returns a FilterFunc which returns true if the mountpoint of a given mount has prefix p, else false.
func HasPrefix(p string) FilterFunc {
  return FilterFunc(func (m *Mount) bool {
    return strings.HasPrefix(m.MountPoint, prefix)
  })
}
```

3. Implement a `Filter` function on Mounts:

```
// Filter returns a filtered copy of Mounts
func (ms Mounts) Filter(f FilterFunc) Mounts {
  filtered := make([]*Mount, 0, len(ms))

  for i, m := range ms {
    if f(m) {
      filtered = append(filtered, m)
    }
  }

  return Mounts(filtered)
}
```

If memory usage is a concern (which here imho is not), we could even use in-place filtering https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating.